Sources/MongoSwift/ChangeStreamEvent.swift

```diff
@@ -1,12 +1,60 @@
 /// An `UpdateDescription` containing fields that will be present in the change stream document for
 /// operations of type `update`.
 public struct UpdateDescription: Codable {
-    /// A document containing key:value pairs of names of the fields that were changed, and the new
-    /// value for those fields.
+    /**
+     * A document containing key:value pairs of names of the fields that were changed (excluding the fields reported
+     * via `truncatedArrays`), and the new value for those fields.
+     *
+     * Despite array fields reported via `truncatedArrays` being excluded from this field, changes to fields of the
+     * elements of the array values may be reported via this field.
+     *
+     * Example:
+     *   original field:
+     *     "arrayField": ["foo", {"a": "bar"}, 1, 2, 3]
+     *   updated field:
+     *     "arrayField": ["foo", {"a": "bar", "b": 3}]
+     *   a potential corresponding UpdateDescription:
+     *     {
+     *       "updatedFields": {
+     *         "arrayField.1.b": 3
+     *       },
+     *       "removedFields": [],
+     *       "truncatedArrays": [
+     *         {
+     *           "field": "arrayField",
+     *           "newSize": 2
+     *         }
+     *       ]
+     *     }
+     *
+     * Modifications to array elements are expressed via dot notation.
+     * Example: an `update` which sets the element with index 0 in the array field named arrayField to 7 is reported as
+     *   "updatedFields": {"arrayField.0": 7}
+     *
+     * - SeeAlso: https://docs.mongodb.com/manual/core/document/#document-dot-notation
+     */
     public let updatedFields: BSONDocument
 
     /// An array of field names that were removed from the document.
     public let removedFields: [String]
+
+    /**
+     * Truncations of arrays may be reported either via this field or via the ‘updatedFields’ field. In the latter
+     * case, the entire array is considered to be replaced. The method used to report a truncation is a server
+     * implementation detail.
+     *
+     * The structure of documents in this field is
+     *   {
+     *      "field": <string>,
+     *      "newSize": <int>
+     *   }
+     *
+     * Example: an `update` which shrinks the array `arrayField.0.nestedArrayField` from size 8 to 5 may be reported
+     * via this field as [{"field": "arrayField.0.nestedArrayField", "newSize": 5}]
+     *
+     * This property will only ever be present on MongoDB server versions >= 5.0.
+     */
+    public let truncatedArrays: [BSONDocument]?
```

Comment:
Is there any particular reason for using just a document here instead of a `Codable` type?

Reply:
nope, good point. I remember having this thought at some point but I never came back to it. I added a new `TruncatedArrayDescription` type.

as an aside, I nested that type in `UpdateDescription`, but it got me thinking we don't seem to nest any other change stream related types that probably could/should be nested, such as `OperationType`.
I opened [SWIFT-1154](https://jira.mongodb.org/browse/SWIFT-1154) about considering that for 2.0.